src/xrpld/app/tx/detail/InvariantCheck.cpp

```diff
@@ -1120,4 +1123,76 @@ ValidMPTIssuance::finalize(
         mptokensCreated_ == 0 && mptokensDeleted_ == 0;
 }
 
+//------------------------------------------------------------------------------
+
+void
+ValidPermissionedDomain::visitEntry(
+    bool,
+    std::shared_ptr<SLE const> const& before,
+    std::shared_ptr<SLE const> const& after)
+{
+    if (after->getType() != ltPERMISSIONED_DOMAIN)
+        return;
+    auto const& credentials = after->getFieldArray(sfAcceptedCredentials);
+    credentialsSize_ = credentials.size();
+    auto const sorted = credentials::makeSorted(credentials);
+    isUnique_ = !sorted.empty();
+
+    if (isUnique_)
+    {
+        unsigned i = 0;
+        for (auto const& cred : sorted)
+        {
+            auto const& credTx(credentials[i++]);
+            isSorted_ = (cred.first == credTx[sfIssuer]) &&
```

Comment:
I don't understand the significance of this check. The list of accepted credentials is already sorted, what is `isSorted_` checking?

Reply:
As I understand Invariants also check incoming transactions / requests

Reply:
okay, I understand it now. the `std::set<>` data structure inherently sorts the pair values. You are validating the input against the `std::set<>` data-structure, alright 👍